### PR TITLE
move to rust 2018

### LINKED ISSUE #172: Rust edition 2018
Is there something blocking the move to rust 2018 ?

I think it would be great to move to this.

To reduce the chance to have conflict, it might be nice to wait until there are no on going PR.

I'm willing to do it if it's ok for you.

### PR DESCRIPTION
**Pull request by antoine-de**
I did not hand clean up all the `extern crate` imports because of lack of motivation (there are tons of them :roll_eyes: ), but I think it can be done later on

related to #172

### CODE REVIEW

**Comment by TeXitoi** [comment]
Did you run `crgo fix --edition-idioms`?

I think #172 can't be closed before extern crate/#[macro_use] cleanup.

**Comment by TeXitoi** [comment]
you need to fmt

File: get-corresponding-derive/src/lib.rs [hunk 1]
```diff
@@ -70,7 +70,7 @@ fn impl_get_corresponding(ast: &syn::DeriveInput) -> quote::Tokens {
             pub trait GetCorresponding<T: Sized> {
                 /// For the given self, returns the set of
                 /// corresponding `T` indices.
-                fn get_corresponding(&self, &#name) -> IdxSet<T>;
+                fn get_corresponding(&self, name: &#name) -> IdxSet<T>;
```
Comment (TeXitoi) [inline get-corresponding-derive/src/lib.rs]: This sould be `model` or `_`.

**Comment by antoine-de** [comment]
oups nop didn't knew about`--edition-idioms` I run only `--edition`

great, it fixes all the `extern crates` but the `macro_use` ones

**Pull request merged**